### DISCUSSION TITLE
Lucia Auth role-based auth

### DISCUSSION THREAD

**Discussion by kernpinoy** [saasykits/nextjs-sessionauth-template#70] [Q&A] [answered]
Hi! This is an interesting starter kit! I think I have some few questions.

How would I do role-based auth in here?

My current way is this:

```
const { user } = await validateRequest();

  if (user) {
    return redirect(`/${user.role}`);
}

```
  
  I do this for every page.tsx of the route.
  
  Now, when compiling, I notice that even my login page is a server function. I have no interactivity in there. It's just a login page.
  
  How do you guys solve this? Thanks!

**Comment by iamtouha** [comment, marked as answer]
Hi, glad to hear that you liked this kit!

Actually the issue you're facing is the expected behavior as `validateRequest` function pulls user session information from `cookies` that makes the whole route dynamic. Also, it is ideal to keep these routes dynamic. otherwise same content will be shown to all visitors whether the have access or not. 

>[cookies()](https://nextjs.org/docs/app/api-reference/functions/cookies) and [headers()](https://nextjs.org/docs/app/api-reference/functions/headers): Using these in a Server Component will opt the whole route into dynamic rendering at request time.

For more information, please read: https://nextjs.org/docs/app/building-your-application/rendering/server-components#dynamic-functions

**Comment by kernpinoy** [comment]
Yeah, that makes a lot of sense. Thanks!
